src/router.rs

```diff
@@ -131,15 +136,27 @@ impl Router {
     /// Route will match any method, including gibberish.
     /// In case of ambiguity, handlers specific to methods will be preferred.
     pub fn any<S: AsRef<str>, H: Handler, I: AsRef<str>>(&mut self, glob: S, handler: H, route_id: I) -> &mut Router {
-        self.mut_inner().wildcard.add(glob.as_ref(), Box::new(handler));
+        self.mut_inner().wildcard.add(glob.as_ref(), Arc::new(handler));
         self.route_id(route_id.as_ref(), glob.as_ref());
         self
     }
 
     fn recognize(&self, method: &method::Method, path: &str)
-                     -> Option<Match<&Box<Handler>>> {
-        self.inner.routers.get(method).and_then(|router| router.recognize(path).ok())
-            .or(self.inner.wildcard.recognize(path).ok())
+                      -> Result<Match<&Arc<Handler>>, IronError> {
```

Comment:
I would recommend using an Enum for all router error types instead of returning `IronError`. So you have an enum 

    enum RouterError {
        MethodNotAllowed,
        NotFound
    }

impl Error for that. Then you don't need that weird hack further down where you compare the debug repr of `IronError.error` with the string `MethodNotAllowed`. Instead you just match an enum.

You could include https://github.com/brson/error-chain as dependency, which automatically generates these kind of errors.